### ISSUE TITLE
Margins within Inset incorrect when dropdown is open

### ISSUE THREAD

**Issue by penx** [radix-ui/themes#169]
When you have the following code:

```jsx
<>
  <Card>
    <Inset>
      <Flex>
        <Box m="1">Item 1</Box>
        <Box m="1">Item 2</Box>
        <Box m="1">Item 3</Box>
      </Flex>
    </Inset>
  </Card>
  <DropdownMenuRoot>
    <DropdownMenuTrigger>
      <Button>Button</Button>
    </DropdownMenuTrigger>
    <DropdownMenuContent>Content</DropdownMenuContent>
  </DropdownMenuRoot>
</>
```

In Chromium (119.0.6045.199), the margin of the items changes when the dropdown is open.

# Closed:

<img width="317" alt="Screenshot 2023-12-01 at 15 08 23" src="https://github.com/radix-ui/themes/assets/1027024/51405fb0-8daa-48b6-82b0-09bda521c268">

<img width="198" alt="Screenshot 2023-12-01 at 15 11 45" src="https://github.com/radix-ui/themes/assets/1027024/a7610e30-313d-41fd-9dc5-0f4a4fe013fb">

# Open:

<img width="315" alt="Screenshot 2023-12-01 at 15 08 27" src="https://github.com/radix-ui/themes/assets/1027024/57ab628b-0df4-43a5-8975-e7e26e0d5f16">

<img width="205" alt="Screenshot 2023-12-01 at 15 11 51" src="https://github.com/radix-ui/themes/assets/1027024/7486062d-9dc9-4ef5-85f0-f2074005848d">

 [Reproducible example](https://penx.github.io/radix-themes-playroom/preview/#?code=N4Igxg9gJgpiBcIA8BhAhgJygPgDoDsACQpASXwGcYAXPI4kgMQBsYAPOhhpAIQjcIBbALy4QARjHZS1GIMLikAej4cCXYr35DREqTLmEATMtWcNWgSLGSQ02fIDMp-uc1KW7c8vJVaBZXQsOiQAEQwIAAcoCAB3fABZGHwAVwAlCAh-ejCI6LjE5JSAFQwASwBzCpgMNxIeFOpqCHxsBqaW00bm1vUSJXComPik1NLK6tq+3KGC0ZSUFtl8WkWV5OplQfyRorXl7K284cLUjKy6EAAaEGoACzk4RAAZSrvqEABfIA) ([editor](https://penx.github.io/radix-themes-playroom/#?code=N4Igxg9gJgpiBcIA8BhAhgJygPgDoDsACQpASXwGcYAXPI4kgMQBsYAPOhhpAIQjcIBbALy4QARjHZS1GIMLikAej4cCXYr35DREqTLmEATMtWcNWgSLGSQ02fIDMp-uc1KW7c8vJVaBZXQsOiQAEQwIAAcoCAB3fABZGHwAVwAlCAh-ejCI6LjE5JSAFQwASwBzCpgMNxIeFOpqCHxsBqaW00bm1vUSJXComPik1NLK6tq%2B3KGC0ZSUFtl8WkWV5OplQfyRorXl7K284cLUjKy6EAAaEGoACzkYCgQAbRAAGUq76hAAXQBfIA))

As far as I can tell, this does not affect Safari/webkit.

**Comment by penx** [comment]
Another example, possibly simpler, that has the same behaviour:

```tsx
<>
  <Box p='2'>
    <Inset>
      <Box>
        <Box m="2">Item 1</Box>
        <Box m="2">Item 2</Box>
      </Box>
    </Inset>
  </Box>
  
  <DropdownMenuRoot>
    <DropdownMenuTrigger>
      <Button>Button</Button>
    </DropdownMenuTrigger>
    <DropdownMenuContent>Content</DropdownMenuContent>
  </DropdownMenuRoot>
</>
```

**Comment by penx** [comment]
Minimal reproduction without using any radix-themes js:

```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <link href="https://www.unpkg.com/@radix-ui/themes@2.0.1/styles.css" rel="stylesheet" />
  </head>
<body>
  <div data-scaling="100%" class="radix-themes">
    <div class="rt-Inset rt-r-side-all rt-r-clip-border-box">
      <div>
        <div class="rt-r-m-8">Item 1</div>
        <div class="rt-r-m-8">Item 2</div>
      </div>
    </div>
  </div>
  <script>setTimeout(() => {
    document.body.style.pointerEvents = "none";
  }, 1000);</script>
</body>
</html>
```

**Comment by penx** [comment]
Minimal reproduction without using any radix themes css or js

```html
<!DOCTYPE html>
<html lang="en">
<head>
  <style>

:where(.rt-Inset) > * {
  --margin-top-override: initial;
}

.rt-Inset:where(.rt-r-side-all) {
  --margin-top-override: 0px;
}

.rt-r-m-8 {
  margin: var(--margin-top-override, 48px);
}
</style>
</head>
<body>
  <div class="rt-Inset rt-r-side-all">
    <div>
      <div class="rt-r-m-8">Item 1</div>
      <div class="rt-r-m-8">Item 2</div>
    </div>
  </div>
  <script>setTimeout(() => {
    document.body.style.pointerEvents = "none";
  }, 1000);</script>
</body>
</html>
```

**Comment by penx** [comment]
Raised upstream issue 

https://bugs.chromium.org/p/chromium/issues/detail?id=1508841

**Comment by penx** [comment]
This was fixed in Chromium version 122.0.6171.0

- https://bugs.chromium.org/p/chromium/issues/detail?id=1508841
- https://chromiumdash.appspot.com/commits?commit=beab1f88a0bddc645e726380a73f0464643e074d&platform=Mac

**Comment by penx** [comment]
I tested the [reproducible example](https://penx.github.io/radix-themes-playroom/preview/#?code=N4Igxg9gJgpiBcIA8BhAhgJygPgDoDsACQpASXwGcYAXPI4kgMQBsYAPOhhpAIQjcIBbALy4QARjHZS1GIMLikAej4cCXYr35DREqTLmEATMtWcNWgSLGSQ02fIDMp-uc1KW7c8vJVaBZXQsOiQAEQwIAAcoCAB3fABZGHwAVwAlCAh-ejCI6LjE5JSAFQwASwBzCpgMNxIeFOpqCHxsBqaW00bm1vUSJXComPik1NLK6tq+3KGC0ZSUFtl8WkWV5OplQfyRorXl7K284cLUjKy6EAAaEGoACzk4RAAZSrvqEABfIA) in Canary 122.0.6251.0 and it's now working fine

**Comment by vladmoroz** [comment]
> Upstream bug was fixed in Chromium version 122.0.6171.0.

Oh interesting, so this was a browser bug after all. Thanks for digging into it
